### ISSUE TITLE
Crashed while trying to end hero turn D2E

### ISSUE THREAD

**Issue by Hobbeslionheart** [NPBruce/valkyrie#568]
It crashed while I was trying to end a turn for a hero on D2E for testing purposes. First time that has happened to me. I'm going to see if I can reproduce it again. Attached is the output log right after the crash.

[output_log.txt](https://github.com/NPBruce/valkyrie/files/1150770/output_log.txt)

**Comment by Hobbeslionheart** [comment]
It crashed again, but it's probably my custom activation that crashed it as the next activation was going to be a lieutenant with a custom activation (same as the first crash). Probably too much text. I will tweak activation to see if that does the trick.

**Comment by Hobbeslionheart** [comment]
My custom monster broke Valkyrie. Once I assigned him some health, everything is fine.

**Comment by NPBruce** [comment]
I will look into this to improve handling, can you please provide the custom monster example with the issues?

**Comment by Hobbeslionheart** [comment]
![image](https://user-images.githubusercontent.com/26637617/28243965-67388d62-69aa-11e7-8eb8-1144b8834909.png)

I think the problem is that health isn't defined when created so when it was time to activate, he technically didn't have any health, but he spawned so that might have confused Valkyrie when he activated as he was Schrodinger's cat (alive and dead at the same time) and crashed it. So a quick workaround would be to have custom monsters have a health of one upon creation.

**Comment by Hobbeslionheart** [comment]
Or rather health is defaulted to 0 when created. 

![image](https://user-images.githubusercontent.com/26637617/28243987-2a89ab2a-69ab-11e7-8885-6f497cd1dbdf.png)

**Comment by NPBruce** [comment]
To clarify, the health number is purely to display bonus health, as per Road to Legend.  Valkyrie does not track health in any way, if this is set to 0 then no bonus health is displayed, which is entirely valid.

Also note that your first image shows a base monster type (Baron Zachareth) and you second does not (entirely new monster) which work quite differently.  In any event a default value of 0 is correct.

I have tried several combinations of configurations and cannot produce the problem you describe.  Affairs of Wizards has a CustomMonster just like this.

It would be helpful if you could attach your scenario in a state where the crash occurs.

**Comment by Hobbeslionheart** [comment]
Unfortunately once i fixed it, I moved on and wasn't able to recreate it. I'll try to start forking my scenarios once game breaking bugs are found. One thing I did notice just now is that I had an event and a token had a display=false in the respective .ini files, but the token popped up fine during spawn although it couldn't reveal the text when clicked on though and the event was just skipped over (although it triggered the next event). I didn't manually add the value in the .ini files (although I manually deleted it once I realized that the token wasn't working) 

I tested the value with the Baron, and it didn't seem to make a difference (I think the portrait didn't show) so I wonder if a random value is being given to a monster in the same way that actually crashed it. None of the other tokens and events I created seem to have that display=false value and I didn't see a way I could have toggled it in the editor for the event and token at least.
